core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -360,13 +389,41 @@ private[spark] class IndexShuffleBlockResolver(
           if (dataTmp != null && dataTmp.exists() && !dataTmp.renameTo(dataFile)) {
             throw new IOException("fail to rename file " + dataTmp + " to " + dataFile)
           }
+
+          // write the checksum file
+          checksumTmpOpt.zip(checksumFileOpt).foreach { case (checksumTmp, checksumFile) =>
+            val out = new DataOutputStream(
+              new BufferedOutputStream(
+                new FileOutputStream(checksumTmp)
+              )
+            )
+            Utils.tryWithSafeFinally {
+              checksums.foreach(out.writeLong)
+            } {
+              out.close()
+            }
+
+            if (checksumFile.exists()) {
+              checksumFile.delete()
+            }
+            if (!checksumTmp.renameTo(checksumFile)) {
+              // It's not worthwhile to fail here after index file and data file are already
+              // successfully stored due to checksum is only used for the corner error case.
+              logWarning("fail to rename file " + checksumTmp + " to " + checksumFile)
```

Comment:
In the `if` condition [above](https://github.com/apache/spark/pull/32401/files#diff-9e9749da596e4dd6c02722f91cd62afc28a44f00c7cebb927ccdeae1629e98a1R354), we will cause index/data to be rewritten even if checksum is not present - while here we do not fail.
Should we make both consistent ?

(If we discussed this earlier, apologies - still catching up on mails).

Reply:
So if you look at the comment, my concern is: it's not worthwhile to fail here after the index file and data file are
successfully generated since the checksum is only a best-effort could do in case of data corruption. 

WDYT?

Reply:
I agree, I am fine with this behavior here.
I was wondering if we have to make it the same above as well ?
That is, if index/data exists but checksum does not - do we want to rewrite index/data just to populate checksum ?
Or simply avoid writing checksum if it is missing and behave like here ?
Thoughts ?

Reply:
I see. I got your point. I'd prefer to back to `if (existingLengths != null) {`.